### DISCUSSION TITLE
Integration with HUnit

### DISCUSSION THREAD

**Discussion by hyperswine** [haskell/haskell-language-server#3464] [Ideas]
I was interested in rust-analyzer's integration with `cargo test` and how quick you can simply write up a unit test and press `run test` above the function to run that specific test. Elm also has something similar which gives you a green tick next to test passes and red crosses next to failures. I think it would be interesting for hls to have something similar.

Also, maybe a way to run functions directly with a `run function` prompt above each function could also allow quick testing of quick inputs in addition to comments with evaluations

**Comment by santiweight** [comment]
I've made good progress here: https://github.com/haskellfoundation/tech-proposals/pull/46

The major things left to do are related to integration with HIE bios so that I get info about available test components and to monitor altered files...

**Reply by hyperswine** [reply]
thats really cool. It seems like you have something thats working pretty well already

**Comment by santiweight** [comment]
I'm definitely interested in any pointers btw wrt hie-bios and also whether such a plugin would work well as part of the `haskell` VSC extensions
